pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -0,0 +1,118 @@
+		return v.validateDelete(req)
+	default:
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("invalid admission webhook request"))
+	}
+}
+
+// InjectDecoder injects the decoder.
+func (v *ConfigMapValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// validateCreate validates the ConfigMap create
+func (v *ConfigMapValidator) validateCreate(req admission.Request) admission.Response {
+	return admission.Allowed("")
+}
+
+// validateUpdate validates the ConfigMap and CloudResourcePrefix update
+func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Response {
+	newConfigMap := &corev1.ConfigMap{}
+	oldConfigMap := &corev1.ConfigMap{}
+	newControllerConfig := &config.ControllerConfig{}
+	oldControllerConfig := &config.ControllerConfig{}
+	err := v.decoder.Decode(req, newConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	} else {
+		if newConfigMap.GetName() != ConfigMap {
+			return admission.Allowed("")
+		}
+		newData := newConfigMap.Data["nephe-controller.conf"]
+		err = yaml.UnmarshalStrict([]byte(newData), newControllerConfig)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	}
+	err = json.Unmarshal(req.OldObject.Raw, &oldConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode old configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	} else {
+		oldData := oldConfigMap.Data["nephe-controller.conf"]
+		err = yaml.UnmarshalStrict([]byte(oldData), oldControllerConfig)
+		if err != nil {
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+	}
+	if newControllerConfig.CloudResourcePrefix == oldControllerConfig.CloudResourcePrefix {
+		v.Log.V(1).Info("CloudResourcePrefix are same", "newData", newControllerConfig.CloudResourcePrefix, "oldData", oldControllerConfig.CloudResourcePrefix)
+		return admission.Allowed("")
+	}
+	// if v.NpController.IsCloudResourceCreated() {
+	//	return admission.Denied(fmt.Sprintf("already cloud resource is created with CloudResourcePrefix %s. Please delete the resource and try or use the same CloudResourcePrefix", oldControllerConfig.CloudResourcePrefix))
+	//}
+	if len(newControllerConfig.CloudResourcePrefix) > 0 {
+		reg := regexp.MustCompile(`^[a-zA-Z0-9]+([_ -]?[a-zA-Z0-9])*$`)
+		valid := reg.MatchString(newControllerConfig.CloudResourcePrefix)
+		if !valid {
+			return admission.Denied(fmt.Sprintf("invalid CloudResourcePrefix %s", newControllerConfig.CloudResourcePrefix))
+		}
+		// Updates the CloudResourcePrefix to the new value
+		v.Config.CloudResourcePrefix = newControllerConfig.CloudResourcePrefix
+	} else {
+		// Updates the CloudResourcePrefix to the default value
+		v.Config.CloudResourcePrefix = config.DefaultCloudResourcePrefix
+	}
+	v.Log.V(1).Info("updated", "CloudResourcePrefix", v.Config.CloudResourcePrefix)
```

Comment:
Info logs should always start with a capital letter.
```suggestion
	v.Log.V(1).Info("Updated", "CloudResourcePrefix", v.Config.CloudResourcePrefix)
```

Reply:
Solved